synapse/storage/databases/main/e2e_room_keys.py

```diff
@@ -271,23 +313,25 @@ async def delete_e2e_room_keys(
         )
 
     @staticmethod
-    def _get_current_version(txn, user_id):
+    def _get_current_version(txn: LoggingTransaction, user_id: str) -> int:
         txn.execute(
             "SELECT MAX(version) FROM e2e_room_keys_versions "
             "WHERE user_id=? AND deleted=0",
             (user_id,),
         )
-        row = txn.fetchone()
-        if not row:
+        row = cast(Tuple[Optional[int]], txn.fetchone())
+        if row[0] is None:
             raise StoreError(404, "No current backup version")
```

Comment:
The previous code was buggy. `SELECT MAX(...)` will return a single row with `NULL` if it's done over 0 rows.

Reply:
oh, because MAX(version) will aggregate to NULL if there are no rows permitted by the where clause? Might that be worth a comment?